src/main/scala/memeid/Time.scala

```diff
@@ -0,0 +1,65 @@
+package memeid.time
+
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.annotation.tailrec
+
+import cats.effect._
+
+trait Time[F[_]] {
+  /* A gregorian time monotonic timestamp. */
+  def monotonic: F[Long]
+}
+
+object Time {
+
+  def tick: Long =
```

Comment:
Can we get this from cats `Clock`?

Reply:
Not really, we are using `tick` in the `AtomicReference#updateAndGet` function to get the current system time and create unique timestamps. The `cats-effect` `Clock` gives us a `F[Long]` when asking for a timestamp, which we'd have to run inside the atomic reference udpate and get function.